instrumentation/instagorillamux/handler_test.go

```diff
@@ -0,0 +1,125 @@
+// (c) Copyright IBM Corp. 2021
+// (c) Copyright Instana Inc. 2016
+
+// +build go1.11
+
+package instagorillamux_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/instana/testify/require"
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/instana/testify/assert"
+
+	instana "github.com/instana/go-sensor"
+	"github.com/instana/go-sensor/instrumentation/instagorillamux"
+
+	"github.com/gorilla/mux"
+)
+
+func TestMain(m *testing.M) {
+	instana.InitSensor(&instana.Options{
+		Service: "gorillamux-test",
+		Tracer: instana.TracerOptions{
+			CollectableHTTPHeaders: []string{"x-custom-header-1", "x-custom-header-2"},
+		},
+	})
+
+	os.Exit(m.Run())
+}
+
+func TestAddMiddleware(t *testing.T) {
+	r := mux.NewRouter()
+	r.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		return
+	})
+	sensor := instana.NewSensor("gorillamux-test")
+
+	assert.Equal(t, 0, getNumberOfMiddlewares(r))
+	instagorillamux.AddMiddleware(sensor, r)
+	assert.Equal(t, 1, getNumberOfMiddlewares(r))
+}
```

Comment:
You are absolutely right. It does not test the functional requirements of the instrumentation. Personally, I found these tests useful, especially at the development phase. To guarantee the correctness of the implementation. Nevertheless, it might be more similar to the testing approach we have in other modules without this test.

Reply:
removed